From c26199b5ca03a677970bb442bafac7aaa039eb70 Mon Sep 17 00:00:00 2001
From: Josh Rabinowitz <joshr@joshr.com>
Date: Thu, 21 Mar 2019 15:18:21 -0400
Subject: [PATCH] respect DESTDIR as per GNU/debian/etc

---
 CHANGELOG.md | 3 ++-
 Makefile     | 5 +++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/CHANGELOG.md b/CHANGELOG.md
index 694952c8..ba8c9aab 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,7 @@
 - Add -v 'verbose' option to 'tell', showing gpg output (#320)
 - Change 'init' to never ignore .secret files (#362)
 - 'add' appends filepaths to .gitignore by default (#225)
+- Automate the GitHub release (#411)
 
 ### Bugfixes
 
@@ -25,7 +26,7 @@
 - Fix re: SECRETS_DIR in 'init' and SECRETS_EXTENSION in test_reveal.bats (#364)
 - Use SECRETS_GPG_COMMAND env var in gpg version check (#389)
 - Add header to git-secret.7 man page, for debian and doc improvement (#386)
-- Automate the GitHub release (#411)
+- Respect DESTDIR when installing as per GNU/debian/etc recommendations (#424)
 
 ## Misc
 
diff --git a/Makefile b/Makefile
index 9f0feb5c..cd5129a8 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,6 @@
 SHELL:=/usr/bin/env bash
 PREFIX?="/usr"
+DESTDIR?=
 
 #
 # Building:
@@ -22,12 +23,12 @@ build: git-secret
 .PHONY: install
 install:
 	chmod +x "./utils/install.sh"; sync; \
-	"./utils/install.sh" "${PREFIX}"
+	"./utils/install.sh" "${DESTDIR}${PREFIX}"
 
 .PHONY: uninstall
 uninstall:
 	chmod +x "./utils/uninstall.sh"; sync; \
-	"./utils/uninstall.sh" "${PREFIX}"
+	"./utils/uninstall.sh" "${DESTDIR}${PREFIX}"
 
 #
 # Testing:
