--- generate-dictionary.sh	2018-03-07 14:50:27.000000000 +0900
+++ generate-dictionary.sh	2018-05-12 12:47:03.340282961 +0900
@@ -6,32 +6,9 @@
 
 
 # ==============================================================================
-# remove tmp files
+# filter entries
 # ==============================================================================
 
-rm -rf ../mozc-$MOZCVER/
-rm -rf ../mozc-neologd-ut-$MOZCVER.$DICVER.$REVISION/
-rm -f ../mozc-neologd-ut-$MOZCVER.$DICVER.$REVISION.tar*
-rm -f chimei/*.zip
-rm -f chimei/*.CSV*
-rm -f chimei/*.py
-rm -f src/id.def
-rm -f src/mecab-user-dict-seed*
-rm -f src/mozcdic*
-rm -f src/neologd*
-rm -f src/zipcode.costs
-
-
-# ==============================================================================
-# extract official mozc and filter entries
-# ==============================================================================
-
-# extract official mozc
-echo "extract official mozc..."
-cd ..
-tar xf mozc-$MOZCVER.tar.bz2
-cd -
-
 echo "filter mozc entries..."
 cd src/
 ruby filter-mozc-entries.rb ../../mozc-$MOZCVER/src/data/dictionary_oss/dictionary00.txt &
@@ -141,15 +118,3 @@
 
 # install fcitx-mozc
 cp -r fcitx-mozc/fcitx ../mozc-$MOZCVER/src/unix/
-
-# install PKGBUILD
-cp PKGBUILD ../mozc-$MOZCVER/
-
-# rename mozc dir to mozc-neologd-ut
-cd ..
-mv mozc-$MOZCVER mozc-neologd-ut-$MOZCVER.$DICVER.$REVISION
-
-# compress mozc-neologd-ut
-echo "compress mozc-neologd-ut..."
-tar cf mozc-neologd-ut-$MOZCVER.$DICVER.$REVISION.tar mozc-neologd-ut-$MOZCVER.$DICVER.$REVISION
-XZ_OPT="--threads=0" xz -f mozc-neologd-ut-$MOZCVER.$DICVER.$REVISION.tar
