--- generate-dictionary.sh.bak	2019-05-27 11:56:32.129655757 +0900
+++ generate-dictionary.sh	2019-05-27 12:11:01.334638619 +0900
@@ -1,37 +1,14 @@
 #!/bin/bash
 
-MOZCVER="2.23.2815.102"
-DICVER="20190422"
-REVISION="1"
+: ${MOZCVER:="2.23.2815.102"}
+: ${DICVER:="20190422"}
+: ${REVISION:="1"}
 
 
 # ==============================================================================
-# remove tmp files
+# filter entries
 # ==============================================================================
 
-rm -rf ../mozc-$MOZCVER/
-rm -rf ../mozc-neologd-ut-$MOZCVER.$DICVER.$REVISION/
-rm -f ../mozc-neologd-ut-$MOZCVER.$DICVER.$REVISION.tar*
-rm -f chimei/*.zip
-rm -f chimei/*.CSV*
-rm -f chimei/*.py
-rm -f src/id.def
-rm -f src/mecab-user-dict-seed*
-rm -f src/mozcdic*
-rm -f src/neologd*
-rm -f src/zipcode.costs
-
-
-# ==============================================================================
-# extract official mozc and filter entries
-# ==============================================================================
-
-# extract official mozc
-echo "extract official mozc..."
-cd ..
-tar xf mozc-$MOZCVER.tar.bz2
-cd -
-
 # get hinsi ID
 cp ../mozc-$MOZCVER/src/data/dictionary_oss/id.def src/
 
@@ -40,13 +17,7 @@
 # generate placenames and ZIP codes
 # ==============================================================================
 
-# get zip code data
-echo "get zip code data..."
 cd chimei/
-wget http://www.post.japanpost.jp/zipcode/dl/kogaki/zip/ken_all.zip
-wget http://www.post.japanpost.jp/zipcode/dl/jigyosyo/zip/jigyosyo.zip
-unzip ken_all.zip
-unzip jigyosyo.zip
 
 # modify zip code data
 echo "modify zip code data..."
@@ -115,18 +86,3 @@
 
 # install fcitx-mozc
 cp -r fcitx-mozc/fcitx ../mozc-$MOZCVER/src/unix/
-
-# install PKGBUILD
-cp PKGBUILD ../mozc-$MOZCVER/
-
-# install a patch
-cp -r patches ../mozc-$MOZCVER/
-
-# rename mozc dir to mozc-neologd-ut
-cd ..
-mv mozc-$MOZCVER mozc-neologd-ut-$MOZCVER.$DICVER.$REVISION
-
-# compress mozc-neologd-ut
-echo "compress mozc-neologd-ut..."
-tar cf mozc-neologd-ut-$MOZCVER.$DICVER.$REVISION.tar mozc-neologd-ut-$MOZCVER.$DICVER.$REVISION
-XZ_OPT="--threads=0" xz -f mozc-neologd-ut-$MOZCVER.$DICVER.$REVISION.tar
