diff --git a/src/nm-l2tp-service.c b/src/nm-l2tp-service.c
index 72a0b3a..df6e921 100644
--- a/src/nm-l2tp-service.c
+++ b/src/nm-l2tp-service.c
@@ -645,7 +645,7 @@ nm_l2tp_config_write (NML2tpPlugin *plugin,
 	/* value = nm_setting_vpn_get_data_item (s_vpn, NM_L2TP_KEY_GATEWAY); */
 	write_config_option (ipsec_fd, "  right=%s\n", priv->saddr);
 	value = nm_setting_vpn_get_data_item (s_vpn, NM_L2TP_KEY_IPSEC_GATEWAY_ID);
-	if(value)write_config_option (ipsec_fd, "  rightid=@%s\n", value);
+	if(value)write_config_option (ipsec_fd, "  rightid=%s\n", value);
 
 	if (!priv->is_libreswan) {
 		write_config_option (ipsec_fd, "  esp=aes128-sha1,3des-sha1\n");
